packages/@react-aria/tag/src/useTag.ts

```diff
@@ -10,74 +10,72 @@
  * governing permissions and limitations under the License.
  */
 
-import {ButtonHTMLAttributes, HTMLAttributes, KeyboardEvent, ReactNode} from 'react';
-import {DOMProps, Removable} from '@react-types/shared';
+import {ButtonHTMLAttributes, HTMLAttributes, KeyboardEvent} from 'react';
 import {filterDOMProps, mergeProps, useId} from '@react-aria/utils';
+import {GridState} from '@react-stately/grid';
 // @ts-ignore
 import intlMessages from '../intl/*.json';
+import {TagProps} from '@react-types/tag';
+import {useGridCell, useGridRow} from '@react-aria/grid';
 import {useMessageFormatter} from '@react-aria/i18n';
 
 
-export interface AriaTagProps extends Removable<ReactNode, void>, DOMProps {
-  children?: ReactNode,
-  isDisabled?: boolean,
-  validationState?: 'invalid' | 'valid',
-  isSelected?: boolean,
-  role?: string
-}
-
 export interface TagAria {
   tagProps: HTMLAttributes<HTMLElement>,
   labelProps: HTMLAttributes<HTMLElement>,
   clearButtonProps: ButtonHTMLAttributes<HTMLButtonElement>
 }
 
-export function useTag(props: AriaTagProps): TagAria {
+export function useTag(props: TagProps<any>, state: GridState<any, any>): TagAria {
```

Comment:
perhaps 
```suggestion
export function useTag<T>(props: TagProps<T>, state: GridState<T, GridCollection<T>>): TagAria {
```
I'll have to double check the types here